ctx.go

```diff
@@ -1054,6 +1055,31 @@ func (c *Ctx) Query(key string, defaultValue ...string) string {
 	return defaultString(c.app.getString(c.fasthttp.QueryArgs().Peek(key)), defaultValue)
 }
 
+// Queries returns a map of query parameters and their values.
+// It returns an error if the query string is not valid.
+func (c *Ctx) Queries() (map[string]string, error) {
+	queryRaw := c.fasthttp.QueryArgs().String()
+	queryRaw, err := url.QueryUnescape(queryRaw)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse query: %w", err)
+	}
+
+	queries, err := url.ParseQuery(queryRaw)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse query: %w", err)
+	}
+
+	m := make(map[string]string)
```

Comment:
I think this can be done with

 ```go
    m := make(map[string]string)

	for k, v := range queries {
		if len(v) > 0 {
			m[k] = v[len(v)-1]
		}
	}
```